### ISSUE TITLE
cpu training is work,cpu training not work

### ISSUE THREAD

**Issue by zhb0920** [pytorch/pytorch#99690]
### 🐛 Describe the bug

i do model=model.cuda().cpu()
cpu training will work

### Versions

1.10
cuda 11.1

**Comment by jgong5** [comment]
Can you check if things work with latest PyTorch? If not, can you also share the detailed error log from the test script? Thanks.

**Comment by zhb0920** [comment]
> Can you check if things work with latest PyTorch? If not, can you also share the detailed error log from the test script? Thanks.

gpu training ，Loss can decrease normally
bug cpu training，Loss cannot decrease normally, but instead rapidly increases
if i do model=model.cuda().cpu()，Loss can decrease normally also。

**Comment by mingfeima** [comment]
@zhb0920 Almost impossible for us to debug with the limited info given in this thread. It is preferred to provide at least a minimal example that can reproduce this issue.

**Comment by zhb0920** [comment]
> @zhb0920 Almost impossible for us to debug with the limited info given in this thread. It is preferred to provide at least a minimal example that can reproduce this issue.
class MyModel
------------------------------------
for gpu model
model=MyModel()
model=model.cuda()
Loss can decrease normally
------------------------------------
for cpu model
if i only do 
model=MyModel()
Loss cannot decrease normally, instead rapidly increases
if i do 
model=MyModel()       model=model.cuda().cpu()
or 
model=MyModel()       model=model.double.float()
Loss can decrease normally
what does those operation do in pytorch may cause the differ?

**Comment by peterbell10** [comment]
@zhb0920 we need the source code for `MyModel` and your training loop, which have the error. Or a simpler script that fails in the same way.
